### PR TITLE
Implement RBAC in admin

### PR DESCRIPTION
**Pull request by rgalanakis**
Overhaul member constraint editing

- Use chips for display
- Allow adding multiple constraints at a time
- Edit the status of the constraints you add

---

Overhaul eligibility constraint components

They were not good UX, so redo them since it's a pretty
common thing admins have to do.

---

Admin: Cleanup global data accesses

Right now we are calling common APIs, like currencies and roles,
in each component. Instead we can cache it in a global context.

Removes the call to get resource access metadata every time
`useResourceAccess` was called (which was most components).

---

Check member role update permissions

We should only be able to modify member roles if we have the
admin management role; otherwise, any admin, like onboarding manager,
can modify any member to make them an admin.

If the admin doesn't have admin management write,
hide the roles from member detail, and prohibit editing.

Also improve the UI for role editing,
so role ordering stays consistent (would jump when adding/removing a role),
labeling is nicer, and code has less duplication.

---

Admin: remove toCreate and toEdit

Instead, standardize resource routes (and support overrides
where the route doesn't match the resource name,
like organization_membership and /membership),
and add canEdit and canCreate props.

---

Add resource-based role access

In the frontend, we need to match the role access for a feature.
So the API and frontend need to use the same knowledge of
'members use the :admin_members role access'.

Build a mapping of resource to role access,
like `Member => [:admin_member, :admin_management]`,
and expose it to the frontend.
Also use this to control the common endpoints,
rather than the endpoint declaring what access it uses
(it does still need to match in one-off endpoints).

Add `useRoleAccess` and `useResourceAccess` hooks,
and hook them up to list and detail pages
(and a few others) to control what is visible.

---

Add support for RBAC into admin

We already had a 'role' model as a very basic form
of RBAC. We only had 'admin', and by its absence, non-admin.

This adds a couple new things:

- Features that describe roles, like 'admin' and 'onboarding manager'.
- A new `Member::RoleAccess` model that encapsulates
  logic about whether a member can perform a certain action.
  This maps things like 'has the onboarding manager role'
  to 'can read/write members but only read offerings'.

---

Update Sentry to pull in session bugfix

See https://github.com/getsentry/sentry-ruby/issues/2268

### CODE REVIEW

File: adminapp/src/pages/OrganizationMembershipForm.jsx [hunk 1]
```diff
@@ -44,7 +44,6 @@ export default function OrganizationMembershipForm({
           helperText={orgText}
           value={resource.verifiedOrganization?.name}
           fullWidth
-          required
```
Comment (rgalanakis) [inline adminapp/src/pages/OrganizationMembershipForm.jsx:47]: Why isn't this required?

File: lib/suma/admin_api/search.rb [hunk 2]
```diff
@@ -158,36 +158,6 @@ def ds_search_or_order_by(column_sym, ds, params)
       present_collection ds, with: SearchTransactionEntity, language: lang
     end
 
-    params do
```
Comment (rgalanakis) [inline lib/suma/admin_api/search.rb:157]: Why was this changed? I don't understand.

File: adminapp/src/pages/OrganizationMembershipForm.jsx (hunk 1 above)
Comment (DeeTheDev) [reply adminapp/src/pages/OrganizationMembershipForm.jsx:47]: @rgalanakis If we want to update other member data and save, the organization field will ask for an input, but it should be allowed to be left empty if we don't want to verify the organization membership.

Comment (rgalanakis) [reply adminapp/src/pages/OrganizationMembershipForm.jsx:47]: This isn't a sufficient fix- it just creates a new bug. We need to be able to modify the name and id separately with proper validation. Please file a bug so we can fix this properly, it should not be here.
